### PR TITLE
[quant][graphmode] Make `InsertQuantDeQuantHelper` global

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30553 [quant][graphmode] `insert_quant_dequant` pass support shared class types
* #30552 [quant][graphmode][refactor] Add registerQParams function
* #30551 [quant][graphmode] Insert GetAttr for quantization parameters instead of Constant
* **#30550 [quant][graphmode] Make `InsertQuantDeQuantHelper` global**

Summary:
Right now we have a `InsertQuantDeQuantHelper` for each module, but we need
it to be global because we need to know what graphs have been quantized before
and based on this information we can decide how to handle the module instance.

Test Plan:
test_jit.py, test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18818651](https://our.internmc.facebook.com/intern/diff/D18818651)

### CODE REVIEW

File: torch/csrc/jit/passes/quantization.cpp [hunk 1]
```diff
@@ -601,37 +605,45 @@ void InsertQuantDeQuantHelper::collectObserverNodesAndValueToQuantize(Value* v)
...
   // Remove observer modules from last one to first one in order to
   // reduce the time complexity, assuming all the observer modules
   // are added after the existing modules, we'll have complexity of
   // O(N) where N is number of observer moduels with this optimization
-  for (int64_t i = observer_modules_to_remove_.size() - 1; i >= 0; --i) {
-    auto observer_name = observer_modules_to_remove_[i];
-    module_._ivalue()->unsafeRemoveAttr(observer_name);
-    module_.type()->unsafeRemoveAttribute(observer_name);
+  if (observer_modules_to_remove_.count(module._ivalue())) {
+    auto observers = observer_modules_to_remove_.at(module._ivalue());
```
Comment (ZolotukhinM) [inline torch/csrc/jit/passes/quantization.cpp]: Please use `const auto&` to avoid copying.

File: torch/csrc/jit/passes/quantization.cpp [hunk 2]
```diff
@@ -851,7 +864,8 @@ script::Module InsertQuantDeQuant(
     const std::string& method_name,
     bool inplace) {
   script::Module module = inplace ? input_module : input_module.clone();
-  InsertQuantDeQuantImpl(module, method_name);
+  InsertQuantDeQuantHelper h;
+  h.run(module, method_name);
 
   // NOTE: Remove observer module does not work right now, we'll return
   // the module with observer modules as a temporary workaround
```
Comment (ZolotukhinM) [inline torch/csrc/jit/passes/quantization.cpp]: This comment is stale btw.

**Review by ZolotukhinM — APPROVED** [review]
Looks good, but please fix clang-format and clang-tidy issues before landing.

**Comment by jerryzh168** [comment]
actually just realized we should use Graph instead of module as key

**Pull request closed without merging**